client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js

```diff
@@ -353,24 +353,31 @@ export default class DeploymentPlugin extends PureComponent {
   }
 
   async getDefaultEndpoint(tab) {
+    const credentials = {
+      clientId: '',
+      clientSecret: '',
+      camundaCloudClientId: '',
+      camundaCloudClientSecret: ''
+    };
+
     let endpoint = {
       id: generateId(),
       targetType: CAMUNDA_CLOUD,
       authType: AUTH_TYPES.NONE,
       contactPoint: '',
       oauthURL: '',
       audience: '',
-      clientId: '',
-      clientSecret: '',
-      camundaCloudClientId: '',
-      camundaCloudClientSecret: '',
       camundaCloudClusterUrl: '',
+      ...credentials,
       rememberCredentials: false
     };
 
     const previousEndpoints = await this.getEndpoints();
     if (previousEndpoints.length) {
-      endpoint = previousEndpoints[0];
+      endpoint = {
+        ...credentials,
+        ...previousEndpoints[0]
+      };
```

Comment:
Is there any way we can test this, to verify the bug is fixed?

Reply:
I will look into adding a test case that ensures this behaviour doesn't happen again

Reply:
I added a test case to check that the default endpoint has the credential keys, when they haven't been saved to the config